src/apps/chifra/pkg/rpcClient/get_receipt.go

```diff
@@ -26,7 +27,18 @@ type ReceiptQuery struct {
 
 // GetTransactionReceipt fetches receipt from the RPC. If txGasPrice is provided, it will be used for
 // receipts in blocks before London
-func GetTransactionReceipt(chain string, query ReceiptQuery) (receipt types.SimpleReceipt, err error) {
+func GetTransactionReceipt(chain string, query ReceiptQuery, cache *cacheNew.Store) (receipt types.SimpleReceipt, err error) {
+	if cache != nil {
+		tx := &types.SimpleTransaction{
+			BlockNumber:      query.Bn,
+			TransactionIndex: query.Txid,
+		}
+		if err := cache.Read(tx, nil); err == nil {
+			// success
+			return *tx.Receipt, nil
```

Comment:
I think in some cases `.Receipt` may be null (in the actual blockchain data), so this may core dump. Only dereference if it's not `nil`.

Reply:
Good point 👍